src/api/system.c

```diff
@@ -162,7 +165,16 @@ static int f_poll_event(lua_State *L) {
       }
 #endif
       lua_pushstring(L, "keypressed");
-      lua_pushstring(L, key_name(buf, e.key.keysym.sym));
+      /* Is the scancode from the keypad and the number-lock off?
+      ** We assume that SDL_SCANCODE_KP_1 up to SDL_SCANCODE_KP_9 and SDL_SCANCODE_KP_0
+      ** and SDL_SCANCODE_KP_PERIOD are declared in SDL2 in that order. */
+      scancode = e.key.keysym.scancode;
+      if (scancode >= SDL_SCANCODE_KP_1 && scancode <= SDL_SCANCODE_KP_1 + 10 &&
+        !(KMOD_NUM & SDL_GetModState())) {
+        lua_pushstring(L, numpad[scancode - SDL_SCANCODE_KP_1]);
+      } else {
+        lua_pushstring(L, key_name(buf, e.key.keysym.sym));
+      }
```

Comment:
I think that this should be done for `SDL_KEYUP` too.
Also, in the issue it looked like this was a Windows problem. Does it happen on other OSs? If not, shouldn't this be under `ifdef`?

Reply:
It is not clear to me if it was a Windows-only problem and the modification looks legit for all OS but this one was just an initial proposition to test. If it is really a windows-only problem we can add the ifdef like in the original proposition.

Reply:
I confirm the problem exists also on linux and the fix seems to work correctly in my tests.

I will see for the KEY_UP for the moment I don't what we are supposed to do, I will have a look.